File: src/IsbInterop/Data/Wrappers/Connection.vb
Imports IsbInterop.Accessory
Imports IsbInterop.Accessory.Wrappers

Namespace Data.Wrappers

  ''' <summary>
  ''' Обертка над IConnection.
  ''' </summary>
  Friend Class Connection
    Inherits IsbComObjectWrapper
    Implements IConnection

    ''' <summary>
    ''' Получает информацию о системе.
    ''' </summary>
    ''' <returns>Информация о системе.</returns>
    Public Function GetSystemInfo() As ISystemInfo Implements IConnection.GetSystemInfo
      Dim rcwSystemInfo = GetRcwProperty("SystemInfo")
      Return New SystemInfo(rcwSystemInfo, Scope)
    End Function

    ''' <summary>
    ''' Конструктор.
    ''' </summary>
    ''' <param name="rcwConnection">СOM-объект IConnection.</param>
    ''' <param name="scope">Область видимости.</param>
    Friend Sub New(rcwConnection As Object, scope As IScope)
      MyBase.New(rcwConnection, scope)
    End Sub
  End Class
End Namespace

File: src/IsbInterop.CSharp/Data/Wrappers/Connection.cs
using IsbInterop.Accessory;
using IsbInterop.Accessory.Wrappers;

namespace IsbInterop.Data.Wrappers
{
  /// <summary>
  /// Обертка над IConnection.
  /// </summary>
  internal class Connection : IsbComObjectWrapper, IConnection
  {
    /// <summary>
    /// Информация о системе.
    /// </summary>
    public ISystemInfo SystemInfo
    {
      get
      {
        var rcwSystemInfo = GetRcwProperty("SystemInfo");
        return new SystemInfo(rcwSystemInfo, Scope);
      }
    }

    /// <summary>
    /// Конструктор.
    /// </summary>
    /// <param name="rcwConnection">СOM-объект IConnection.</param>
    /// <param name="scope">Область видимости.</param>
    internal Connection(object rcwConnection, IScope scope)
      : base(rcwConnection, scope) { }
  }
}
